### PR TITLE
fix(grpo): respect independent token and mask strides

### PR DESCRIPTION
**Pull request by LOGO127**
## Summary

Fix the default Triton selective-log-softmax helper using the token-ID row stride to address the mask and assuming unit column strides for both inputs.

The helper independently slices token IDs and the mask to the trailing completion length. Those views can have different row/column strides, including zero strides from expansion. The old kernel can silently select the wrong mask entry or token ID. Pass each view's two strides to the kernel; keep logit-contiguity requirements, output layout, temperature scaling and loss math unchanged. No new input materialization is introduced.

## Testing Done

- Hardware Type: NVIDIA A100-80G-PCIe.
- Base: 95b01e94027cc70ffec7233e89d3e9fa31fb72f7.
- New CUDA regression: 61 failed / 14 passed on unchanged production code; 75 passed after the fix.
- Final combined run with the existing GRPO suite: 179 passed / 17 existing unsupported-loss-combination skips. This includes the 75 new cases, not 179 additional cases.
- Entire-repository Ruff lint and format checks passed (ruff 0.15.7, 467 formatted files). These are the read-only checks used by checkstyle; the auto-fixing Make target was not invoked.
- Environment: Python 3.10, PyTorch 2.3.0a0+ebedce2, CUDA runtime 12.3, Triton 3.3.1 in a task-local dependency directory. No global Torch/Triton packages were changed. Tests ran with repository conftest active and third-party pytest plugin autoload disabled.
- Dense-input forward microbenchmark: tiny [3,8,2053] input increased from 38.80us to 44.14us; [4,257,32000] was 51.28us to 51.88us; [1,1025,64000] was 91.88us to 91.63us. Five alternating-order rounds, ten calls/round. This is a correctness fix, not a speedup claim.

- [ ] run `make test` to ensure correctness (scoped suites above only)
- [ ] run `make checkstyle` to ensure code style (read-only Ruff lint/format checks passed)
- [ ] run `make test-convergence` to ensure convergence (not run)

No full-model convergence, distributed training, AMD, NPU or non-Triton backend validation is claimed. Dense inputs with matching strides were not shown to be incorrect. The reproducer uses extra backing capacity to demonstrate incorrect values without out-of-bounds accesses.

### Reproduction commands and retained output

The CUDA runs used the source checkout with task-local Triton 3.3.1 first on `PYTHONPATH`, `PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python`, and `PYTEST_DISABLE_PLUGIN_AUTOLOAD=1` (repository conftest remained active). From that checkout:

```bash
python3 -m pytest test/transformers/test_selective_log_softmax_strides.py test/transformers/test_grpo_loss.py -q
```

```text
179 passed, 17 skipped, 1 warning in 2.92s
```

The warning is the unused `asyncio_mode` configuration with third-party plugin autoload disabled. Skips are the existing unsupported LUSPO/VESPO combinations, not missing CUDA. These results were generated before publication; the unchanged production/test files and JUnit counts were rechecked for submission, and repository-wide lint/format checks were rerun.

AI assistance: implementation, tests, and PR drafting.

### CODE REVIEW

**Comment by LOGO127** [comment]
Could a maintainer review the independent token/mask stride handling at b9f6fd1 and advise which upstream validation lane is appropriate? There are currently no check results listed on this PR, so I am not treating the published local evidence as upstream CI.

The A100 regression and reproduction command are in the description: 179 passed / 17 existing unsupported-combination skips, including the 75 added cases. The small dense-input overhead is also reported there. This remains a correctness fix for independently sliced/expanded views, not a general speedup or a full-model convergence claim. Thanks!
